### PR TITLE
Pywebsocket refactoring

### PR DESCRIPTION
**Pull request by arenevier**

### CODE REVIEW

**Comment by arenevier** [comment]
@ricea

File: tools/wptserve/wptserve/ws_h2_handshake.py [hunk 1]
```diff
@@ -8,39 +8,21 @@
...
 def check_connect_method(request):
     if request.method != u'CONNECT':
         raise HandshakeException('Method is not CONNECT: %r' % request.method)
 
 
-class WsH2Handshaker(object):
+class WsH2Handshaker(HandshakerBase):
     def __init__(self, request, dispatcher):
-        """Opening handshake processor for the WebSocket protocol (RFC 6455).
+        """Bootstraping handshake processor for the WebSocket protocol with HTTP/2 (RFC 8441).
```
Comment (ricea) [inline tools/wptserve/wptserve/ws_h2_handshake.py:25]: "Bootstrapping"

**Review by ricea — COMMENTED** [review]
lgtm

I verified the `tools/third_party/pywebsocket3` files are identical to upstream commit [265579a969e1aec2a6457db7674a9e31fbb7d564 ](https://github.com/GoogleChromeLabs/pywebsocket3/commit/265579a969e1aec2a6457db7674a9e31fbb7d564).

**Comment by arenevier** [comment]
Are the tests failure on windows legitimate failures?
https://dev.azure.com/web-platform-tests/b14026b4-9423-4454-858f-bf76cf6d1faa/_apis/build/builds/64487/logs/203

I don't have a windows machine. But on linux, I can't reproduce when running
$ tox -c tools/wpt/ -e py38 -- --junitxml=results.xml

**Review by LukeZielinski — APPROVED** [review]
Check failures are unrelated (#28745)

**Comment by foolip** [comment]
@arenevier can you try rebasing? Those CI failures should be resolved now.

Also, everyone on this PR might be interested in https://github.com/web-platform-tests/wpt/issues/28863, which I filed after noticing we're going in an amusing direction :)

**Comment by arenevier** [comment]
> @arenevier can you try rebasing? Those CI failures should be resolved now.


Done.

I have separeted in two commits: first the merge obtained with

```git subtree pull --prefix tools/third_party/pywebsocket3 https://github.com/GoogleChromeLabs/pywebsocket3 master `--squash```

Then, the actual changes to wpt.

I'm not sure if it would be better to squash them.

**Comment by foolip** [comment]
In order to preserve the history correctly and make future `git subtree` calls work correctly it will be necessary to merge this into wpt with branch protection temporarily disabled.

Right @stephenmcgruer, is that how you've been doing it?

**Comment by ricea** [comment]
Ping?

**Comment by foolip** [comment]
@jgraham let me know on IRC that the branch protection rules don't apply to admins, which I can confirm in settings. Makes sense, that's the same reason why admin merging in the UI is allowed. (Force pushing or deleting `master` is still not allowed for admins, fortunately.)

I'll go ahead and merge this from the command line.

**Pull request merged**